### PR TITLE
fix: use sorted breakpoints in useBreakpoint

### LINKED ISSUE #5555: useBreakpointValue() works incorrectly with custom breakpoints
### Description

I define additional breakpoint value preSm: '36em'. It works properly with style properties, but when I use useBreakpointValue({ base: true, lg: false }), it returns true for base to preSm and md to lg, but false for preSm to md

### Link to Reproduction

https://codesandbox.io/s/adoring-sky-khuko?file=/src/index.js

### Steps to reproduce

1. Add custom breakpoint to theme (preSm: '36em')
2. Use useBreakpointValue({ base: true, lg: false })
3. Get true for:
base to preSm
md to lg
4. Get false for:
preSm to md
lg+

It fixes by adding preSm to responsive object - useBreakpointValue({ base: true, preSm: true,  lg: false })

### Chakra UI Version

1.6.7

### Browser

all

### Operating System

- [X] macOS
- [X] Windows
- [X] Linux

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by TimKolberger**
Closes #5555 (not yet. The CSS responsive object uses media min queries, where as the useBreakpoint uses minmax queries)

## 📝 Description

Fixed an issue where the hook `useBreakpoint` did not work as expected with custom breakpoints

## ⛳️ Current behavior (updates)

useBreakpoint used user defined values from the theme. It is operating on an unsorted object.

## 🚀 New behavior

It uses the processed and sorted breakpoints from `toCSSVar(theme)`.

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/media-query/src/use-breakpoint.ts [hunk 1]
```diff
@@ -92,11 +92,6 @@ export function useBreakpoint(defaultBreakpoint?: string) {
       // push the media query list handleChange
       // so we can use it to remove Listener
       listeners.add({ mediaQuery, handleChange })
-
-      return () => {
-        // clean up 1
-        mediaQuery.removeListener(handleChange)
-      }
```
Comment (TimKolberger) [inline packages/media-query/src/use-breakpoint.ts:99]: The return value is unused in `.forEach`, so I removed it.

**Review by cschroeter — APPROVED** [review]
LGTM

**Review by nikolovlazar — APPROVED** [review]
👌

File: packages/media-query/src/use-breakpoint.ts (hunk 1 above)
Comment (TimKolberger) [reply packages/media-query/src/use-breakpoint.ts]: I rewrote the whole thing now. It was "historically grown".

**Comment by TimKolberger** [comment]
Just before merging I reviewed the file again, and I did not understand immediately what was going on. So I rewrote the whole hook and reused the breakpoint logic from the theme breakpoints. And I fixed a bug on the way there.

Can I request a review again @nikolovlazar and @cschroeter? 💖

**Review by nikolovlazar — APPROVED** [review]

**Pull request merged**